src/instrumentations/extentions/http.ts

```diff
@@ -75,25 +75,75 @@ function createHttpRequestHook(
       return;
     }
 
+    const headers = !(request instanceof IncomingMessage)
+      ? request.getHeaders()
+      : request.headers;
+    for (const headerKey in headers) {
+      const headerValue = headers[headerKey];
+
+      if (headerValue === undefined) {
+        continue;
+      }
+      span.setAttribute(
+        `http.request.header.${headerKey.toLocaleLowerCase()}`,
+        headerValue
+      );
+    }
+
+    /* TODO: add body capture
     if (request instanceof IncomingMessage) {
-      // TODO: add attributes here
+      // request body capture
+      const listener = (chunk: any) => {
+        console.log('Dataaa: ', chunk);
+      };
+
+      request.on('data', listener);
+      request.once('end', () => {
+        request.removeListener('data', listener);
+      });
     }
+     */
   };
 }
 
 function createHttpResponseHook(
   options: Options
 ): HttpResponseCustomAttributeFunction {
   return (span, response) => {
-    if (!(response instanceof ServerResponse)) {
-      return;
-    }
-
     const spanContext = span.spanContext();
 
     if (!isSpanContextValid(spanContext)) {
       return;
     }
-    // TODO: add attributes here
+
+    const headers = !(response instanceof IncomingMessage)
```

Comment:
this seems a bit of logical complexity, why checking for "not" instead of switching the order of ? a:b value?

Reply:
Good point, fixed :)